src/scrape_up/imdb1/imdb.py

```diff
@@ -0,0 +1,127 @@
+
+                if self.name and movie_name.lower() == self.name.lower():
+                    return f"Rank: {rank}\nYear: {year}\nRating: {rating}"
+
+            if self.name:
+                return "Movie not found in the top-rated list."
+
+            return movie_data
+
+        except requests.exceptions.RequestException as e:
+            return f"Error: {e}"
+
+    def save_to_excel(self):
+        movie_data = self.details()
+        if not movie_data:
+            return "No movie data available."
+
+        excel = openpyxl.Workbook()
+        sheet = excel.active
+        sheet.title = "Top Rated Movies"
+
+        """
+        Sheet will have 4 cols 
+
+        """
+        sheet.append(['Movie Rank', 'Movie Name', 'Year of Release', 'IMDB Rating'])
+
+        if isinstance(movie_data, list):
+            for movie in movie_data:
+                sheet.append(movie)
+        else:
+            movie = movie_data.split('\n')
+            sheet.append([movie[0].split(': ')[1], self.name, movie[1].split(': ')[1], movie[2].split(': ')[1]])
+
+        filename = "IMDB Ratings.xlsx"
+        excel.save(filename)
+        return f"Movie data saved to {filename}"
+    
+
+    def scrape_movies(self):
+        try:
+            source = requests.get("https://www.imdb.com/chart/top/?ref_=nv_mv_250")
+            source.raise_for_status()
+
+            soup = BeautifulSoup(source.text, 'html.parser')
+            movies = soup.find('tbody', class_="lister-list").find_all('tr')
+
+            """
+            Extracting the details of the Top 250 movies. 
+
+            """
+
+            movie_data = []
+            for movie in movies:
+                movie_name = movie.find('td', class_="titleColumn").a.text
+                rank = movie.find('td', class_="titleColumn").get_text(strip=True).split('.')[0]
+                year = movie.find('td', class_="titleColumn").span.text.strip('()')
+                rating = movie.find('td', class_="ratingColumn imdbRating").strong.text
+
+                movie_data.append([rank, movie_name, year, rating])
+
+            return movie_data
+
+        except requests.exceptions.RequestException as e:
+            print(f"Error: {e}")
+            return []
+
+
+    """
+    Creating an excel file that has the list of all 250 movies.
+
+    """
+
+    def save_all_movies_to_excel(self):
+        movie_data = self.scrape_movies()
+        if not movie_data:
+            return "No movie data available."
+
+        excel = openpyxl.Workbook()
+        sheet = excel.active
+        sheet.title = "Top Rated Movies"
+        sheet.append(['Movie Rank', 'Movie Name', 'Year of Release', 'IMDB Rating'])
+
+        for movie in movie_data:
+            sheet.append(movie)
+
+        filename = "IMDB Ratings.xlsx"
+        excel.save(filename)
+        return f"Movie data saved to {filename}"
+
+movie_name = input("Enter a movie name (optional): ")
+movie = Movie(name=movie_name)
```

Comment:
Great work @Shreyg-27 | 
Update the `documentation.md` file about the class and how to use it. You can see other classes mentioned in the same file for reference.